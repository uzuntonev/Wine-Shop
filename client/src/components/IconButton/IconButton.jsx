import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { IconButton as IconButtonMaterial } from '@material-ui/core';
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
import {
  ShoppingBasket,
  Delete,
  Edit,
  MoreHoriz,
  Remove,
  Add,
  PhotoCamera,
} from '@material-ui/icons';

const IconButton = ({ handler, icon, attr, children }) => {
  const mapIcon = {
    edit: <Edit />,
    delete: <Delete />,
    more: <MoreHoriz />,
    cart: <AddShoppingCartIcon />,
    basket: <ShoppingBasket />,
    remove: <Remove />,
    add: <Add />,
    camera: <PhotoCamera />,
  };
  const handleIcon = useMemo(() => mapIcon[icon], [icon]);
  return (
    <IconButtonMaterial {...attr} onClick={handler}>
      {handleIcon}
      {children}
    </IconButtonMaterial>
  );
};

IconButton.propTypes = {
  handler: PropTypes.func,
  icon: PropTypes.string.isRequired,
  attr: PropTypes.object,
};

export default IconButton;
