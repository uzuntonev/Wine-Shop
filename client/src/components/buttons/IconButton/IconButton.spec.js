import React from 'react';
import IconButton from './IconButton';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
test('Component "IconButton" is mount', () => {
  const component = renderer.create(
    <BrowserRouter>
      <IconButton handler={() =>{}} icon={'add'} />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
