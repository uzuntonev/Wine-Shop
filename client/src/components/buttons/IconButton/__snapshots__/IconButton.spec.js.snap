// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component "IconButton" is mount 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </span>
</button>
`;
